postproc/caburst.py

```diff
@@ -1,114 +1,154 @@
+        for op in ["max", "min"]:
+            traces_STEPS4.append(
+                Trace(
+                    f"{membrane} {op} V",
+                    "mV",
+                    multi=multi,
+                    reduce_ops={
+                        "amin": [],
+                        "amax": [],
+                        "['i_peak_t', 0]": [],
+                        "['i_peak_y', 0]": [],
+                        # "['i_peak_y', 1]": [],
+                        "['val', 0.038]": [],
+                        "n_peaks": [],
+                    },
+                )
             )
-        )
-traces_STEPS3.append(Trace("t", "ms", multi=multi))
-
-
-""" create the benchmark database"""
-STEPS3_DB = TraceDB(
-    "STEPS3",
-    traces_STEPS3,
-    "caburst/raw_traces/STEPS3",
-    clear_raw_traces_cache=True,
-    clear_refined_traces_cache=True,
-)
-
-
-"""Create the sample traces"""
-multi = 1000
-traces_STEPS4 = []
-
-for membrane in ["smooth", "spiny"]:
-    for op in ["max", "min"]:
-        traces_STEPS4.append(
-            Trace(
-                f"{membrane} {op} V",
-                "mV",
-                multi=multi,
-                reduce_ops={
-                    "amin": [],
-                    "amax": [],
-                    "['i_peak_t', 0]": [],
-                    "['i_peak_y', 0]": [],
-                    # "['i_peak_y', 1]": [],
-                    "['val', 0.038]": [],
-                    "n_peaks": [],
-                },
+    traces_STEPS4.append(Trace("t", "ms", multi=multi))
+
+    """Create the sample database"""
+    STEPS4_DB = TraceDB(
+        "STEPS4",
+        traces_STEPS4,
+        STEPS4_raw_traces_folder,
+        clear_raw_traces_cache=True,
+        clear_refined_traces_cache=True,
+    )
+
+    """Create the comparator for advanced studies"""
+    comp = Comparator(traceDBs=[STEPS3_DB, STEPS4_DB])
+
+    """Perform the ks test"""
+    filter = []
+    for tDBnames, ks_tests in comp.test_ks(filter=filter).items():
+        print(tDBnames)
+        for t, d in sorted(ks_tests.items(), key=lambda t: Utils.natural_keys(t[0])):
+            for k, v in sorted(d.items(), key=lambda k: Utils.natural_keys(k[0])):
+                print(t, k, v)
+
+    """Plots"""
+
+    # fig, ax = plt.subplots(2, 2)
+    # subplot_label = "A"
+    # for i, membrane in enumerate(["smooth", "spiny"]):
+    #     for j, op in enumerate(["max", "min"]):
+    #         comp.avgplot_raw_traces(
+    #             trace_name=f"{membrane} {op} V",
+    #             conf_lvl=0,
+    #             savefig_path="caburst/pics",
+    #             suffix="",
+    #             pplot=ax[i][j],
+    #             legendfontsize=5,
+    #         )
+    #         ax[i][j].set_title(subplot_label + "\n", loc="left", fontweight="bold")
+    #         subplot_label = chr(ord(subplot_label) + 1)
+    # fig.tight_layout()
+    # Figure.savefig(savefig_path=savefig_path, file_name="avg_and_std", fig=fig)
+    # fig.show()
```

Comment:
ehm... yes and no. The problem is that the frequency analysis works only when we use SI units otherwise it is bugged. I tried a few times but there is something that I am not nailing and when I work with neuron units this graph become extremely long to build (and also is unreadable since the bins become too small). Until now I left it commented and uncommented it only when I was using SI units